lib/libutils/ext/trace.c

```diff
@@ -91,6 +117,12 @@ void trace_printf(const char *function, int line, int level, bool level_ok,
 		return;
 	boffs += res;
 
+	/* Print the core ID if in atomic context  */
+	res = print_core_id(buf + boffs, sizeof(buf) - boffs);
+	if (res < 0)
+		return;
+	boffs += res;
```

Comment:
IMO, should be conditional on `CFG_MSG_LONG_PREFIX_MASK` so that INFO messages don't show it by default (similar to thread id). And https://github.com/OP-TEE/optee_os/blob/master/mk/config.mk#L76 needs updating.